src/Riok.Mapperly/Descriptors/MappingBuilders/EnumToStringMappingBuilder.cs

```diff
@@ -18,6 +20,47 @@ public static class EnumToStringMappingBuilder
 
         // to string => use an optimized method of Enum.ToString which would use slow reflection
         // use Enum.ToString as fallback (for ex. for flags)
-        return new EnumToStringMapping(ctx.Source, ctx.Target, ctx.SymbolAccessor.GetAllFields(ctx.Source));
+        return BuildEnumToStringMapping(ctx);
+    }
+
+    private static EnumToStringMapping BuildEnumToStringMapping(MappingBuilderContext ctx)
+    {
+        var ignoredSourceMembers = ctx.Configuration.Enum.IgnoredSourceMembers.ToHashSet(SymbolTypeEqualityComparer.FieldDefault);
+        var explicitMappings = BuildExplicitValueMappings(ctx);
+
+        ctx.AddUnmatchedSourceIgnoredMembers(ignoredSourceMembers);
+        return new EnumToStringMapping(ctx.Source, ctx.Target, ctx.SymbolAccessor.GetAllFields(ctx.Source), explicitMappings);
+    }
+
+    private static IReadOnlyDictionary<IFieldSymbol, AttributeValue> BuildExplicitValueMappings(MappingBuilderContext ctx)
+    {
+        var explicitMappings = new Dictionary<IFieldSymbol, AttributeValue>(SymbolEqualityComparer.Default);
+        foreach (var (source, target) in ctx.Configuration.Enum.ExplicitMappings)
+        {
+            var sourceField = source.ConstantValue.GetEnumField()!;
+
+            if (!SymbolEqualityComparer.Default.Equals(source.ConstantValue.Type, ctx.Source))
+            {
+                ctx.ReportDiagnostic(
+                    DiagnosticDescriptors.SourceEnumValueDoesNotMatchSourceEnumType,
+                    sourceField,
+                    source.ConstantValue.Value ?? 0,
+                    source.ConstantValue.Type?.ToDisplayString() ?? "unknown",
+                    ctx.Source
+                );
+                continue;
+            }
+
+            if (explicitMappings.ContainsKey(sourceField))
```

Comment:
Nitpick: I just noted there is the dictionary extension method `TryAdd` which does exactly this (`ContainsKey` + `Add`), no else would be needed then.

Reply:
Yes, agreed. I have made this change.